src/Extension/UrlExtension.php

```diff
@@ -47,43 +47,8 @@ public function __construct(UrlHelper $urlHelper, ServerUrlHelper $serverUrlHelp
      */
     public function register(Engine $engine) : void
     {
-        $engine->registerFunction('url', [$this, 'generateUrl']);
-        $engine->registerFunction('serverurl', [$this, 'generateServerUrl']);
-        $engine->registerFunction('route', [$this, 'getRouteResult']);
-    }
-
-    /**
-     * Get the RouteResult instance of UrlHelper, if any.
-     */
-    public function getRouteResult() : ?RouteResult
-    {
-        return $this->urlHelper->getRouteResult();
-    }
-
-    /**
-     * Generate a URL from either the currently matched route or the specfied route.
-     *
-     * @param array  $options Can have the following keys:
-     *     - router (array): contains options to be passed to the router
-     *     - reuse_result_params (bool): indicates if the current RouteResult
-     *       parameters will be used, defaults to true
-     * @return string
-     */
-    public function generateUrl(
-        string $routeName = null,
-        array $routeParams = [],
-        array $queryParams = [],
-        ?string $fragmentIdentifier = null,
-        array $options = []
-    ) {
-        return $this->urlHelper->generate($routeName, $routeParams, $queryParams, $fragmentIdentifier, $options);
-    }
-
-    /**
-     * Generate a fully qualified URI, relative to $path.
-     */
-    public function generateServerUrl(string $path = null) : string
```

Comment:
This is a BC break, since it removes public methods, even if they're not used internally.

Instead, please keep them, but add the notation `@deprecated since 2.2.0; to be removed in 3.0.0. These methods were used originally internally when registering functions with the Plates engine; we now register the (URL|Server URL) helper instance directly.`. 

This also means you'll need to rebase this against the `develop` branch and update the pull request to indicate it's against that branch.

Reply:
right, they were marked "public"...